### PR TITLE
Add API to MS

### PR DESCRIPTION
**Pull request by palango**
This adds a REST API to the MS.

This is done in preparation for https://github.com/raiden-network/raiden-services/issues/7 for which we need a working API to export prometheus stats. However, it's also useful for https://github.com/raiden-network/raiden-services/issues/501, https://github.com/raiden-network/raiden-services/issues/650 and https://github.com/raiden-network/raiden-services/issues/763

The changes include:
- `mypy` fixes to changes caused by ne `__init__.py` file
- Make `MonitoringService` a `greenlet`
- Add `MsApi` class with currently only a `/info` endpoint similar to the PFS'
- Add an API test
- Move common test stuff to libs

### CODE REVIEW

File: src/pathfinding_service/cli.py [hunk 1]
```diff
@@ -128,7 +130,7 @@ def main(  # pylint: disable=too-many-arguments,too-many-locals
             raise Exception("PFS did not start within time.")
 
         log.debug("Starting API")
-        api = ServiceApi(
+        api = PfsApi(
```
Comment (karlb) [inline src/pathfinding_service/cli.py:133]: I would have expected `PFSApi`, but both are a bit awkward in their own way.

File: src/monitoring_service/api.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                "service_token_address": to_checksum_address(self.service_token_address),
+                "confirmed_block": {
+                    "number": self.monitoring_service.context.ms_state.blockchain_state.latest_committed_block  # noqa
+                },
+            },
+            "version": self.version,
+            "contracts_version": self.contracts_version,
+            "operator": self.api.operator,
+            "message": self.api.info_message,
+            "UTC": datetime.utcnow().isoformat(),
```
Comment (karlb) [inline src/monitoring_service/api.py:51]: What are the security implications of providing this information? If it is public, attackers might watch for `latest_committed_block` to be outdated and take advantage of that situation.

File: src/monitoring_service/service.py [hunk 3]
```diff
@@ -78,7 +78,8 @@ def handle_event(event: Event, context: Context) -> None:
                 sentry_sdk.capture_exception(ex)
 
 
-class MonitoringService:  # pylint: disable=too-few-public-methods,too-many-instance-attributes
+class MonitoringService(gevent.Greenlet):
```
Comment (karlb) [inline src/monitoring_service/service.py:81]: Please don't inherit from `Greenlet`! It ruins mypy checks and couples the implementation to gevent.

We didn't settle on gevent guidelines yet, but I suggest something like
```python

ms_greenlet = spawn_named("MonitoringService", ms)
...
ms_greenlet.kill()
```

File: src/monitoring_service/api.py (hunk 2 above)
Comment (palango) [reply src/monitoring_service/api.py:51]: There might be, even though I'm not aware of any as the MS are not giving any guarantees right now. However, I'd keep the API non-public for now.

File: src/monitoring_service/cli.py [hunk 4]
```diff
@@ -80,24 +113,40 @@ def main(  # pylint: disable=too-many-arguments
...
+
+        if debug_shell:
+            import IPython
+
+            IPython.embed()
+            return 0
 
-    if debug_shell:
-        import IPython
+        task = gevent.spawn(service.start)
```
Comment (karlb) [inline src/monitoring_service/cli.py:136]: Why not spawn_named?

File: src/monitoring_service/cli.py [hunk 5]
```diff
@@ -80,24 +113,40 @@ def main(  # pylint: disable=too-many-arguments
...
 
-    ms.start()
+        task.get()
+    finally:
+        log.info("Stopping Monitoring Service...")
+        if api:
+            api.stop()
+        if task:
+            task.kill()
+            task.join()
```
Comment (karlb) [inline src/monitoring_service/cli.py:149]: Wouldn't `task.get()` be even better here? If it stops due to a GreenletExit  generated by `kill`, it should pass without an error, but raise errors that happened during cleanup.

File: src/monitoring_service/api.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    def run(self, host: str, port: int) -> None:
+        self.rest_server = WSGIServer((host, port), self.flask_app)
+        self.rest_server.start()
+
+        log.info("Running endpoint", endpoint=f"http://{host}:{port}")
+
+    def stop(self) -> None:
+        if self.rest_server:
+            self.rest_server.stop()
```
Comment (karlb) [inline src/monitoring_service/api.py:91]: If we want to go all in with the `spawn/kill` model, this should be
```python
    def run(self, host: str, port: int) -> None:
        self.rest_server = WSGIServer((host, port), self.flask_app)
        log.info("Running endpoint", endpoint=f"http://{host}:{port}")
        self.rest_server.server_forever()
```
But we can start doing that when we have some gevent guidelines.

Comment (palango) [reply src/monitoring_service/api.py:91]: Let's do that later...

**Pull request merged**